app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,46 @@
+# Comments model belongs to book and to user.
+class CommentsController < ApplicationController
+  def show
+    comment = Comment.find(params[:id])
+    render json: comment, status: :ok
+  end
+
+  def index
+    comments = Comment.where(book_id: params[:book_id])
+    render json: comments, status: :ok
+  end
+
+  def create
+    authorize :comment
+    book = Book.find(params[:book_id])
+    comment = book.comments.build(comments_params)
+    comment.user_id = current_user.id
+    if comment.save
+      render json: comment, status: :ok
+    else
+      render json: { errors: comment.errors }, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    authorize :comment
+    comment = current_user.comments.find(params[:id])
+    if comment.update(comments_params)
+      render json: comment, status: :ok
+    else
+      render json: { errors: comment.errors }, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    authorize :comment
```

Comment:
pass the comment to the authorize method, only admins and the comment's owner should be able to delete it!